lib/dgram.js

```diff
@@ -232,7 +232,9 @@ Socket.prototype.bind = function(port_, address_ /* , callback */) {
     this.on('listening', onListening);
   }
 
-  if (port instanceof UDP) {
+  if (port !== null &&
+      typeof port === 'object' &&
+      typeof port.recvStart === 'function') {
```

Comment:
Perhaps a more foolproof check might be to check for `port[owner_symbol]`?

Reply:
I’m not sure how that check would look like (in a way that still works)?

If you do want something more foolproof, I guess we could add an extra method in `UDPWrapBase::AddMethods()` that would be used specifically for brand checks.

Reply:
I'm going to go ahead and mark this resolved. If there are improvements to be made here they can be done in a separate PR